File: examples/example-56/ruby/features/step_definitions/http.rb
Given('an HTTP {string} call {string} is made') do |method, url|
  execute_http_request({
                           url: url,
                           method: method,
                           headers: {},
                           body: nil
                       })
end

Then('the HTTP call should have returned an HTTP {int}') do |status|
    expect(HttpTest.get_responses[0].code).to eq(status.to_s)
end

File: examples/example-56/js/features/step_definitions/http.js
const { Before, Given, Then } = require('@cucumber/cucumber');
const { executeHttpRequest } = require('../../utils/http');
const assert = require('assert');


Before({ tags: '@http-ignore-errors' }, function () {
    this.http = this?.http ?? {};
    this.http.ignoreErrors = true;
});

Given('an HTTP {string} call {string} is made', async function(method, url) {
    const response = await executeHttpRequest(this, { method, url });
});

Then('the HTTP call should have returned an HTTP {int}', function (status) {
    assert.strictEqual(this?.http?.responses?.[0]?.status, status);
});
